### ISSUE TITLE
catch catches error in monad continuation

### ISSUE THREAD

**Issue by StillerHarpo** [Gabriel439/turtle#421]
I have the following code
```haskell
prog1 :: Shell ()
prog1 = do
  echo "prog1"
  catch
    ( do
        echo "will be catched"
        pure ())
    (\(_ :: SomeException) -> do
        echo "exception")

prog2 :: IO ()
prog2 = do
  echo "prog2"
  procs "lll" [] mempty

mainMenu' :: Shell ()
mainMenu' = do
  echo "mainMenu"
  prog1
  liftIO prog2
```
I would expect that running this will not run the error handler (it will not print "exception). But runnig it prints it
```haskell
*Main> sh $ mainMenu'
mainMenu
prog1
will be catched
prog2
exception
prog2
*** Exception: lll: createProcess: posix_spawnp: does not exist (No such file or directory)
```
Doing the same thing in the IO monad works as I expect

**Comment by Gabriel439** [comment]
Yeah, there's not a good way to fix this, short of removing the `MonadCatch` instance altogether.

What's happening is that whenever you `catch` something in `Shell` the exception handler is catching all exceptions that will ever be raised by that `Shell`, not just exceptions in the wrapped block.

**Comment by denar50** [comment]
@Gabriel439 regarding the error itself `createProcess: posix_spawnp: does not exist (No such file or directory)`, how can I fix this? it doesn't happen on my linux machine, but when running this program in an Amazon Linux image it breaks with that error.
